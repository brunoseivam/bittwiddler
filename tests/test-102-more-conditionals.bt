main {
    var x : int16 = 0;
    if x == 0 {
        emit("x is 0\n");
    } else {
        emit("x is not 0\n");
    };

    var y : int32 = -100;

    var sign_of_y : int32 = if y > 0 { 1; } elif y < 0 { -1; } else { 0; };
    emit('sign of {y} is {sign_of_y}\n');

    var forty_two : uint32 = 0x42;

    if forty_two == 0 {
        emit("forty_two==0\n");
    } elif forty_two == 42 {
        emit("forty_two==42\n");
    } elif forty_two == 0x42 {
        emit("forty_two==0x42\n");
    };
}
