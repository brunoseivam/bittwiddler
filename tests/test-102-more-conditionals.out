x is 0
sign of -100 is -1
forty_two==0x42
